### ISSUE TITLE
Statevec and stabilizer evotype TypeErrors in replib

### ISSUE THREAD

**Issue by sserita** [pyGSTio/pyGSTi#161]
**Describe the bug**
The `algorithms/advanced/ForwardSimulationTypes` and `objects/advanced/ProcessorSpec` tutorials currently fail with the following error:

```
Exception ignored in: 'pygsti.objects.replib.fastreplib.convert_rhoreps'
Traceback (most recent call last):
  File "/Users/sserita/Documents/repos/pyGSTi/pygsti/objects/mapforwardsim.py", line 218, in _bulk_fill_probs_block
    replib.DM_mapfill_probs_block(self, array_to_fill, slice(0, array_to_fill.shape[0]),  # all indices
TypeError: Cannot convert pygsti.objects.replib.fastreplib.SBStateRep to pygsti.objects.replib.fastreplib.DMStateRep
```

There seems to be a missing set of logic that switches the underlying `replib` calls for the different evotypes. PyGSTi 0.9.7 seems to contain some of the missing logic, which was then deprecated in favor of `replib`.

**To Reproduce**
1. Checkout the `develop` branch (or downstream branches such as `feature-tutorial-fix`)
2. Run `algorithms/advanced/ForwardSimulationTypes` or `objects/advanced/ProcessorSpec` tutorials

**Expected behavior**
Tutorials run properly (excluding other minor syntax changes from `develop` branch).

**Environment (please complete the following information):**
 - pyGSTi commit `30ee52d` (on `feature-tutorial-fix` branch)
 - python 3.8.6
 - OSX 10.14.6

**Comment by sserita** [comment]
This issue is mostly fixed with the evotype refactor. One remaining item before complete resolution:

- [ ] Add unitary propagation to densitymx evotype

**Comment by enielse** [comment]
This is outdated, but the functionality should be fixed, in version 0.9.10.  I believe this is because FullUnitaryOp objects can be created using the `densitymx` evolution type.
